src/phoenix/server/api_key_validator.py

```diff
@@ -0,0 +1,65 @@
+from asyncio import create_task, sleep
+from typing import Any, Dict, cast
+
+import jwt
+from jwt import InvalidTokenError
+from sqlalchemy import and_, select
+
+from phoenix.auth import JwtPayload
+from phoenix.config import PHOENIX_SECRET
+from phoenix.db import models
+from phoenix.server.types import DaemonTask, DbSessionFactory
+
+
+class ApiKeyValidator(DaemonTask):
+    def __init__(
+        self,
+        db: DbSessionFactory,
+        sleep_seconds: int = 1,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self._db = db
+        self._seconds = sleep_seconds
+        self._cache: Dict[int, models.APIKey] = {}
+
+    def __call__(self, token: str) -> bool:
+        try:
+            payload = cast(JwtPayload, jwt.decode(token, PHOENIX_SECRET, algorithms=["HS256"]))
+        except InvalidTokenError:
+            return False
+        return (
+            (id_ := payload.get("id_")) is not None
+            and (key := self._cache.get(id_)) is not None
+            and key.name == payload.get("name")
+            and key.description == payload.get("description")
+            and key.created_at.timestamp() == payload.get("iat")
+            and (int(key.expires_at.timestamp()) if key.expires_at else None) == payload.get("exp")
```

Comment:
Check the token has not expired?

Reply:
Aside from the ID check to verify the API key has not been revoked, what makes the other checks necessary?

Reply:
> Check the token has not expired?

This is already checked by `jwt`.

> what makes the other checks necessary?

I don't know. I'm just checking them because they're in the payload: why else would they need to be in the payload?